flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/Fabric8FlinkKubeClient.java

```diff
@@ -130,15 +132,16 @@ public void createJobManagerComponent(KubernetesJobManagerSpecification kubernet
 
     @Override
     public CompletableFuture<Void> createTaskManagerPod(KubernetesPod kubernetesPod) {
```

Comment:
We also need to add a test to guard this change.

Reply:
I was also thinking about adding some tests but just didn't have an idea that how I could test that one variable is moved from local to global. Or maybe it is already covered by the existing tests (such as Fabric8FlinkKubeClientTest#testCreateFlinkTaskManagerPod() )?

Reply:
I think we could create two TM pods and verify that getting the jobmanager deployment should only happen once.